frontend_tests/node_tests/stream_data.js

```diff
@@ -529,3 +529,42 @@ zrequire('stream_data');
     assert(!stream_data.get_default_status('remove_me'));
     assert.equal(page_params.realm_default_streams.length, 0);
 }());
+
+(function test_create_sub() {
+    stream_data.clear_subscriptions();
+    var india = {
+        stream_id: 102,
+        name: 'India',
+        subscribed: true,
+    };
+
+    var canada = {
+        name: 'Canada',
+        subscribed: true,
+    };
+
+    var antarctica = {
+        stream_id: 103,
+        name: 'Antarctica',
+        subscribed: true,
+        color: '#76ce90',
+    };
+
+    var india_sub = stream_data.create_sub_from_server_data('India', india);
+    assert(india_sub);
+    assert.notEqual(india_sub.color, undefined);
+    assert.equal(stream_data.create_sub_from_server_data('India', india), india_sub); // make sure sub doesn't get created twice
+
+    var called = false;
+    global.blueslip.fatal = function (msg) {
+        assert.equal(msg, 'We cannot create a sub without a stream_id');
+        called = true;
+    };
+    var ok = stream_data.create_sub_from_server_data('Canada', canada);
+    assert.equal(ok, undefined);
+    assert(called);
+
+    var antarctica_sub = stream_data.create_sub_from_server_data('Antarctica', antarctica);
```

Comment:
What I was thinking here, was a `sub` that didn't have a color and testing that a color is being assigned.

One thing you can do is make sure that `stream_color.pick_color` is called and override it by making it return a set value. Then asserting that color for the `sub`. Does that make sense?

Because right now, you cover all the lines, but you never assert a `sub` color where the sub color wasn't already defined beforehand.